pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/localbalancerset_controller.go

```diff
@@ -0,0 +1,513 @@
+	currentPoolServices, err := r.currentPoolServices(svc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get current pool services for service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	if canRemoveFinalizer(currentPoolServices) {
+		if err := r.removeFinalizer(svc); err != nil {
+			return errors.Wrapf(err, "failed to add finalizer")
+		}
+		return nil
+	}
+
+	if err := r.deletePoolServices(currentPoolServices); err != nil {
+		return errors.Wrapf(err, "failed to delete all pool services for service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	return nil
+}
+
+func (r *ReconcilePoolService) currentPoolServices(svc *corev1.Service) ([]netv1alpha1.PoolService, error) {
+	listSelector := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			network.LabelServiceName: svc.Name,
+			labelManageBy:            names.LoadBalancerSetController}),
+		Namespace: svc.Namespace,
+	}
+
+	poolServiceList := &netv1alpha1.PoolServiceList{}
+	if err := r.List(context.TODO(), poolServiceList, listSelector); err != nil {
+		return nil, errors.Wrapf(err, "failed to list pool service with %s/%s", svc.Namespace, svc.Name)
+	}
+
+	currentPoolServices := filterInvalidPoolService(poolServiceList.Items)
+
+	return currentPoolServices, nil
+}
+
+func filterInvalidPoolService(poolServices []netv1alpha1.PoolService) []netv1alpha1.PoolService {
+	var filteredPoolServices []netv1alpha1.PoolService
+	for _, item := range poolServices {
+		if !isValidPoolService(&item) {
+			klog.Warningf("Pool Service %s/%s is not valid, label %s or %s is modified",
+				item.Namespace, item.Name, network.LabelServiceName, network.LabelNodePoolName)
+			continue
+		}
+		filteredPoolServices = append(filteredPoolServices, item)
+	}
+	return filteredPoolServices
+}
+
+func isValidPoolService(poolService *netv1alpha1.PoolService) bool {
+	if poolService.Labels == nil {
+		return false
+	}
+	if poolService.Labels[network.LabelServiceName]+"-"+poolService.Labels[network.LabelNodePoolName] != poolService.Name {
+		return false
+	}
+	return true
+}
+
+func canRemoveFinalizer(poolServices []netv1alpha1.PoolService) bool {
+	return len(poolServices) == 0
+}
+
+func (r *ReconcilePoolService) deletePoolServices(poolServices []netv1alpha1.PoolService) error {
+	for _, ps := range poolServices {
+		if !ps.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		if err := r.Delete(context.Background(), &ps); err != nil {
+			return errors.Wrapf(err, "failed to delete poolservice %s/%s", ps.Namespace, ps.Name)
+		}
+	}
+	return nil
+}
+
+func (r *ReconcilePoolService) reconcileNormal(svc *corev1.Service) error {
+	if err := r.addFinalizer(svc); err != nil {
+		return errors.Wrapf(err, "failed to add finalizer")
+	}
+
+	currentPoolServices, err := r.currentPoolServices(svc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get current pool services for service %s/%s", svc.Namespace, svc.Name)
+	}
+
+	if err := r.aggregateToService(svc, currentPoolServices); err != nil {
```

Comment:
aggregateToService should be removed out of reconcileNormal, and add a new function named reconcileService for updating service status and annotations.

Reply:
done